### ISSUE TITLE
TensorFlow lite micro on Cortex-M7 

### ISSUE THREAD

**Issue by gajendrahatt** [tensorflow/tensorflow#72364]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.13.1

### Custom code

No

### OS platform and distribution

S32K344 Cortex-M7

### Mobile device

Cortexm-M7

### Python version

3.9

### GCC/compiler version

arm-none-eabi-gcc 10.2

### Current behavior?

I trained the tensor-flow model with layers Conv2d followed bye relu and Maxpool2D. 
this is the architecture i used
import tensorflow as tf
model1 = tf.keras.Sequential()
# model1.add(tf.keras.Input(shape=(750,3,1)))

model1.add(Conv2D(4,kernel_size=(5,5),strides=(1,1), padding='same',input_shape=(750, 3, 1)))
model1.add(tf.keras.layers.Activation(activation='relu'))
model1.add(MaxPooling2D(pool_size=(4,4),padding='same'))   
# model1.add(Dropout(0.25))

model1.add(Conv2D(8,kernel_size=(5,5),padding='same'))
model1.add(tf.keras.layers.Activation(activation='relu'))
model1.add(MaxPooling2D(pool_size=(4,4), padding='same'))  

model1.add(Conv2D(8,kernel_size=(5,5),padding='same'))
model1.add(tf.keras.layers.Activation(activation='relu'))
model1.add(MaxPooling2D(pool_size=(4,4), padding='same'))  

model1.add(Flatten())
model1.add(Dense(7))
model1.add(tf.keras.layers.Activation(activation='softmax'))
model1.compile(optimizer=tf.keras.optimizers.Adam(learning_rate=0.0004518074511548236754), loss='sparse_categorical_crossentropy', metrics=['accuracy'])
model1.summary()

training and testing  accuracy is more than 95 %.
i convert the model to tflite using
converter = tf.lite.TFLiteConverter.from_keras_model(q_aware_model)
tflite_quant_model = converter.convert()

and the make it in flattebuffers model.h using xxd -i model.tflite > model.h

and created the inference in c++ code for tflite micro

#include "tensorflow/lite/micro/kernels/micro_ops.h"

#include "tensorflow/lite/micro/micro_error_reporter.h"

#include "tensorflow/lite/micro/micro_interpreter.h"

#include "tensorflow/lite/micro/micro_mutable_op_resolver.h"

#include "tensorflow/lite/schema/schema_generated.h"

#include "tensorflow/lite/c/common.h"

#include "tensorflow/lite/micro/micro_profiler.h"

// Include the model data

#include "new_model_data.h"

 
#define INPUT_SIZE 2250

#define OUTPUT_SIZE 7
 
 
// Calculates size of model 118380
 
constexpr int kTensorArenaSize = 10208*sizeof(float);
 
// Globals, used for compatibility with Arduino-style sketches.

namespace {

tflite::MicroErrorReporter micro_error_reporter;

tflite::ErrorReporter* error_reporter = nullptr;

const tflite::Model* model = nullptr;

tflite::MicroInterpreter* interpreter = nullptr;

TfLiteTensor* input = nullptr;

TfLiteTensor* output = nullptr;

uint8_t tensor_arena[kTensorArenaSize];

tflite::MicroProfiler profiler;

}  // namespace
 
void setup() {

  // Set up logging.

  error_reporter = &micro_error_reporter;

  error_reporter->Report("model test on cortexm7");
 
  model = tflite::GetModel(Conv2d_model_v3_f16_tflite);

  if(model==nullptr)

  {

	  printf("model errror");

  }
 
  if (model->version() != TFLITE_SCHEMA_VERSION) {

    TF_LITE_REPORT_ERROR(error_reporter,

                         "Model provided is schema version %d not equal "

                         "to supported version %d.",

                         model->version(), TFLITE_SCHEMA_VERSION);

    return;

  }
 
  printf("getting the model \n");
 
   // This pulls in all the operation implementations we need.

  static tflite::MicroMutableOpResolver<5> micro_op_resolver;

  micro_op_resolver.AddConv2D(tflite::Register_CONV_2D());

  micro_op_resolver.AddMaxPool2D(tflite::Register_MAX_POOL_2D());

  micro_op_resolver.AddRelu();

//  micro_op_resolver.AddReshape();

  micro_op_resolver.AddFullyConnected(tflite::Register_FULLY_CONNECTED());

  micro_op_resolver.AddSoftmax(tflite::Register_SOFTMAX());
 
  // Build an interpreter to run the model with.

  static tflite::MicroInterpreter static_interpreter(

      model, micro_op_resolver, tensor_arena, kTensorArenaSize,nullptr,&profiler);

  interpreter = &static_interpreter;
 
  // Allocate memory from the tensor_arena for the model's tensors.
 
  printf("allocating the tesnors \n ");
 
  TfLiteStatus allocate_status = interpreter->AllocateTensors();
 
  if (allocate_status != kTfLiteOk) {

     MicroPrintf("AllocateTensors() failed");

     return;

   }

  	  input = interpreter->input(0);
 
}

void loop() {
 
  	  printf("data initialisation\n");

	  float input_data[INPUT_SIZE] = {};
printf("allocating data to model \n");

//	  memcpy(input->data.f16, &input_data, 2250*sizeof(float));

		 for (int i = 0; i < INPUT_SIZE; i++) {

			 input->data.f16[i] = input_data[i];

		  }
 
	  if (kTfLiteOk != interpreter->Invoke()) {

	     MicroPrintf("Invoke failed.");

	   }

	  TfLiteTensor* output = interpreter->output(0);
 
 
  // Read the output

	  for (int i = 0; i < OUTPUT_SIZE; ++i) {

    float output_value = output->data.f[i];

    error_reporter->Report("Output value[%d]: %f", i, output_value);

  }

}
 
int main() {
 
  setup();

  while (true) {

    loop();

  }

  return 0;

} 
Now the problem is that when i feeding the input it is showing in the output, means it is pointing the same memory location for input tensor and output tensor.
what is the the feasible solution for this.

### Standalone code to reproduce the issue

```shell
.
```

### Relevant log output

```shell
1221 input the output also 2212
```

**Comment by pkgoogle** [comment]
Hi @gajendrahatt, I believe https://github.com/tensorflow/tflite-micro/issues will be able to assist you better. Thanks.
